contracts/Pricer.sol

```diff
@@ -352,4 +345,88 @@ contract Pricer is OpsManaged, PricerInterface {
+    /// @param _beneficiary beneficiary
+    /// @param _transferAmount transferAmount
+    /// @param _commissionBeneficiary commissionBeneficiary
+    /// @param _commissionAmount commissionAmount
+    /// @return bool isValid
+    function isValidBeneficiaryData(
+        address _beneficiary,
+        uint256 _transferAmount,
+        address _commissionBeneficiary,
+        uint256 _commissionAmount)
+        internal
+        returns (bool /*isValid*/)
+    {
+        require(_beneficiary != address(0));
+        require(_transferAmount != 0);
+
+        if (_commissionAmount > 0) {
+            require(_commissionBeneficiary != address(0));
+        }
+        return true;
+    }
+
+    /// @dev    Takes _spender, _beneficiary, _tokenAmount, _commissionBeneficiary, _commissionTokenAmount;
+    ///         Perform tokenAmount transfer
+    ///         Perform commissionTokenAmount transfer
+    ///         internal method;
+    /// @param _spender spender
+    /// @param _beneficiary beneficiary
+    /// @param _tokenAmount tokenAmount
+    /// @param _commissionBeneficiary commissionBeneficiary
+    /// @param _commissionTokenAmount commissionTokenAmount
+    /// @return (bool)
+    function performTransfer(
+        address _spender,
+        address _beneficiary,
+        uint256 _tokenAmount,
+        address _commissionBeneficiary,
+        uint256 _commissionTokenAmount)
+        internal
+        returns (
+            bool /* boolean value */)
+    {
+        require(EIP20Interface(pricerBrandedToken).transferFrom(_spender, _beneficiary, _tokenAmount));
+
+        if (_commissionBeneficiary != address(0)) {
+            require(EIP20Interface(pricerBrandedToken).transferFrom(_spender,
+            _commissionBeneficiary, _commissionTokenAmount));
+        }
+        return true;
+    }
+
+    /// @dev    Takes _currency, _intendedPricePoint, _transferAmount, _commissionAmount;
+    ///         Validate accepted margin
+    ///         Calculates tokenAmount and commissionTokenAmount
+    ///         internal method
+    /// @param _currency currency
+    /// @param _intendedPricePoint intendedPricePoint
+    /// @param _transferAmount transferAmount
+    /// @param _commissionAmount commissionAmount
+    /// @return (pricePoint, tokenAmount, commissionTokenAmount)
+    function validateMarginAndCalculateBTAmount(
+        bytes3 _currency,
+        uint256 _intendedPricePoint,
+        uint256 _transferAmount,
+        uint256 _commissionAmount)
+        internal
+        returns (uint256, uint256, uint256) /* pricePoint, tokenAmount, commissionTokenAmount */
+    {
+        uint256 pricePoint = getPricePoint(_currency);
+        require(pricePoint > 0);
+        require(isPricePointInRange(_intendedPricePoint, pricePoint, acceptedMargins(_currency)));
+
+        uint256 tokenAmount;
+        uint256 commissionTokenAmount;
+        (tokenAmount, commissionTokenAmount) = getBTAmountFromCurrencyValue(pricePoint, _transferAmount,
```

Comment:
this function call to `getBTAmountFromCurrencyValue` should be inlined (as demonstrated that it calls again for `getPricePoint`; and the function itself on L221 should be removed;

the functionality of having this function (L221) public is achievable in the interaction layer by using `getPricePoint`;

Reply:
as we concluded getPricePoint is called once in pay flow

Reply:
correction, `getBTAmountFromCurrencyValue` does not query `getPricePoint`; there is still value in looking to inline this function to reduce the stack depth;

in return `getPricePointandCalculateAmounts` can be moved to the interaction layer

Reply:
@benjaminbollen: are you noting this for posterity or commenting that this is a change you would like Abhay to make (which I think contradicts what we agreed in the meeting)? And does this relate to #50?